### ISSUE TITLE
Purchase Order: Saving not possible after deleting an Item

### ISSUE THREAD

**Issue by zongo811** [frappe/erpnext#34843]
### Information about bug

I created a new Purchase Order, added 2 Items and realized I just needed 1. I deleted the additional item and tried to save the new Purchase Order. Bute there was no reaction from the Form when I clicked on "Save".

To reproduce:
1. Create new Purchase Order, fill mandatory fields
2. Add 2 Items
3. Delete Item in Row 2
4. Press "Save"
5. Switch to "Developer Tools > Console" and feast your eyes upon the error that just appeared.

### Module

buying

### Version

erpnext 14.20.2
frappe 14.31.0
hrms 14.2.1

### Installation method

manual install

### Relevant log output / Stack trace / Full Error Message.

```shell
TypeError: locals[doc.doctype][doc.name] is undefined
    is_empty_row save.js:67
    remove_empty_rows save.js:74
    remove_empty_rows save.js:58
    save save.js:19
    save save.js:269
    validate_and_save form.js:807
```

**Comment by dvdl16** [comment]
Same for us:

ERPNext: v14.21.0
Frappe Framework: v14.32.1

Installation method: Docker

![image](https://user-images.githubusercontent.com/4974009/231980484-4cf3594c-3937-425e-add1-859d33e6705f.png)

**Comment by precihole** [comment]
ERPNext: v14.21.0 (HEAD)
Frappe Framework: v14.32.1 (HEAD)

I am also facing same issue

**Comment by precihole** [comment]
This Issue comes in all Doctype while amending and deleting an item.

**Comment by dvdl16** [comment]
I've replicated this on BOM's as well...

1. On an existing BOM, click "New Version"
2. Don't Save
3. Add a new BOM Item line
4. Delete the Line
5. Try to Save

**Comment by s-aga-r** [comment]
@zongo811 unable to replicate:

[Screencast from 2023-05-15 10-34-23.webm](https://github.com/frappe/erpnext/assets/63660334/e8deef89-a9d8-4d89-91c9-250659423a32)

**Comment by s-aga-r** [comment]
@dvdl16 unable to replicate on BOM as well:

[Screencast from 2023-05-15 10-37-10.webm](https://github.com/frappe/erpnext/assets/63660334/621af65d-af81-48c2-8595-78d806dc22f6)

**Comment by dvdl16** [comment]
@s-aga-r Can you try it like this, without actually populating the second row with an item code?


https://github.com/frappe/erpnext/assets/4974009/38532936-041b-4531-93c3-fc48332c3257

**Comment by s-aga-r** [comment]
> @s-aga-r Can you try it like this, without actually populating the second row with an item code?
> 
>  20230515.PO.mp4

@dvdl16

[Screencast from 2023-05-15 11-13-16.webm](https://github.com/frappe/erpnext/assets/63660334/c8c33de6-6277-4ade-b69b-1b4661f45fc9)

**Comment by zongo811** [comment]
After updating to
erpnext 14.24.1
frappe 14.36.0

the error does not occur anymore. I have an instance running with
erpnext 14.23.4
frappe 14.35.0

where I can reproduce the error as described above and shown by dvdl16. 

@dvdl16: please "bench update" if possible. The error seems to be fixed with frappe 14.36.0.

**Comment by dvdl16** [comment]
@zongo811 Thanks a ton! I'll update our sites

**Comment by zongo811** [comment]
@precihole please "bench update" if possible. The error seems to be fixed with frappe 14.36.0.
Can you confirm this?

**Comment by precihole** [comment]
Yes it is solved in that release
